lib/internal/fs/recursive_watch.js

```diff
@@ -154,14 +154,11 @@ class FSWatcher extends EventEmitter {
             this.#symbolicFiles.add(f);
           }
 
+          this.#files.set(f, file);
```

Comment:
We don't need line 181 `if (existingStat)` check any more, since existingStat will always exist.

Reply:
removing it breaks existing tests - it is still needed
```console
./node test/parallel/test-fs-watch-recursive.js                            ✔  5s 
node:internal/fs/recursive_watch:181
      if (!existingStat.isDirectory() &&
                        ^

TypeError: Cannot read properties of undefined (reading 'isDirectory')
    at StatWatcher.<anonymous> (node:internal/fs/recursive_watch:181:25)
    at StatWatcher.emit (node:events:519:28)
    at StatWatcher.onchange (node:internal/fs/watchers:89:8)
```

Reply:
👍